tests/System.Text.Primitives.Tests/Encoding/Utf8Utf16ConversionTests.cs

```diff
@@ -78,6 +78,127 @@ public void EnumerateAndEnsureCodePointsOfTheSameUtf8AndUtf16StringsAreTheSame(s
             }
         }
 
+        public static object[][] InvalidUtf8ToUtf16SequenceData = {
```

Comment:
These tests are specific to going from UTF-8 to UTF-16. You'd have to craft different tests for other conversions.

Most of these tests are generic to benefit any implementation. However, the "long" versions know that the buffer needs to be greater than 13 bytes to get into the "fast" path. So, there are some tests specific to ensure we exercise all the code.